src/easyscience/Objects/variable/descriptor_array.py

```diff
@@ -0,0 +1,760 @@
+
+        :return: variance.
+        """
+        return self._array.variances
+
+    @variance.setter
+    @property_stack_deco
+    def variance(self, variance: Union[list, np.ndarray]) -> None:
+        """
+        Set the variance of self. Ensures the input is an array and matches the shape of the existing values.
+
+        :param variance: New variance for the DescriptorArray, must be a list or numpy array.
+        """
+        if variance is not None:
+            if not isinstance(variance, (list, np.ndarray)):
+                raise TypeError(f"{variance=} must be a list or numpy array.")
+            if isinstance(variance, list):
+                variance = np.array(variance)  # Convert lists to numpy arrays for consistent handling.
+
+            if variance.shape != self._array.values.shape:
+                raise ValueError(f"{variance=} must have the same shape as the array values.")
+
+            if not np.all(variance >= 0):
+                raise ValueError(f"{variance=} must only contain non-negative values.")
+
+        self._array.variances = variance
+        
+    @property
+    def error(self) -> Optional[np.ndarray]:
+        """
+        The standard deviations, calculated as the square root of variances.
+
+        :return: A numpy array of standard deviations, or None if variances are not set.
+        """
+        if self._array.variances is None:
+            return None
+        return np.sqrt(self._array.variances)
+
+    @error.setter
+    @property_stack_deco
+    def error(self, error: Union[list, np.ndarray]) -> None:
+        """
+        Set the standard deviation for the parameter, which updates the variances.
+
+        :param error: A list or numpy array of standard deviations.
+        """
+        if error is not None:
+            if not isinstance(error, (list, np.ndarray)):
+                raise TypeError(f"{error=} must be a list or numpy array.")
+            if isinstance(error, list):
+                error = np.array(error)  # Convert lists to numpy arrays for consistent handling.
+
+            if error.shape != self._array.values.shape:
+                raise ValueError(f"{error=} must have the same shape as the array values.")
+
+            if not np.all(error >= 0):
+                raise ValueError(f"{error=} must only contain non-negative values.")
+
+            # Update variances as the square of the errors
+            self._array.variances = error**2
+        else:
+            self._array.variances = None
+
+    def convert_unit(self, unit_str: str) -> None:
+        """
+        Convert the value from one unit system to another.
+
+        :param unit_str: New unit in string form
+        """
+        if not isinstance(unit_str, str):
+            raise TypeError(f'{unit_str=} must be a string representing a valid scipp unit')
+        try:
+            new_unit = sc.Unit(unit_str)
+        except UnitError as message:
+            raise UnitError(message) from None
```

Comment:
no additional context for this reraise - please consider either adding more info or let it percolate up

Reply:
I removed the try catch block. I took the block from `DescriptorNumber`, but it seems like `scipp` has been updated so this is not needed anymore.